### PR TITLE
environment(): Allow stacking append() and prepend()

### PR DESCRIPTION
**Pull request by xclaesse**

### CODE REVIEW

**Comment by jpakkane** [comment]
Anyone want to guess how much breakage this change will cause? (Or, in other words, should there be a deprecation release first?)

**Comment by xclaesse** [comment]
It has been warning for years already, I think it should be safe.

File: docs/markdown/Reference-manual.md [hunk 1]
```diff
@@ -2739,8 +2739,19 @@ tests and other functions. It has the following methods.
   joined by the separator, e.g.  `env.set('FOO', 'BAR'),` sets envvar
   `FOO` to value `BAR`. See `append()` above for how separators work.
 
-**Note:** All these methods overwrite the previously-defined value(s)
-if called twice with the same `varname`.
+*Since 0.58.0* `append()` and `prepend()` methods can be called multiple times
+on the same `varname`. Earlier Meson versions would warn and only the last
+opperation was taking effect.
```
Comment (QuLogic) [inline docs/markdown/Reference-manual.md:2744]:
```suggestion
operation took effect.
```

File: docs/markdown/snippets/environment.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
+## Multiple append() and prepend() in `environment()` object
+
+`append()` and `prepend()` methods can now be called multiple times
+on the same `varname`. Earlier Meson versions would warn and only the last
+opperation was taking effect.
```
Comment (QuLogic) [inline docs/markdown/snippets/environment.md:5]:
```suggestion
operation took effect.
```

**Comment by xclaesse** [comment]
@jpakkane @dcbaker any objection on this?

**Comment by jpakkane** [comment]
It has warned that the behaviour is bad but it has not warned that it will change in the next release. If it had been prohibited before there would be no issue. Not sure if it really matters, but experience has shown that these kinds of changes have regression potential. :(

**Comment by xclaesse** [comment]
I'll probably regret saying this, but I'm sure it's going to be fine... :D

**Pull request merged**